index.js

```diff
@@ -105,34 +118,46 @@ class PearIPC extends ReadyResource {
     this.stream.on('close', onclose)
 
     this._register()
+
+    if (this.server === null && this.id === -1) {
+      this._heartbeat = setInterval(() => {
+        this._beat()
+      }, HEARTBEAT_INTERVAL)
+      await this._beat()
+    }
+  }
+
+  async _beat () {
+    let result = null
+    try { result = await this._ping() } catch { this.close() }
```

Comment:
Detail: there's some duplication in this code: if ping throws, result is null, so this.close() will be called anyway on the next line

Reply:
@HDegroote that's okay - ReadyResource close is designed to be called multiple times - the conditional for handling that is in ReadyResource so no need to handle it here - keeping the check means it will also close on unexpected output where there is no error